tests/cxx/iwyu_stricter_than_cpp.cc

```diff
@@ -61,6 +62,15 @@ void TestTypedefs() {
   // IWYU: IndirectStruct2 is...*iwyu_stricter_than_cpp-i2.h
   DoubleTypedef dt(6);
 
+  // "indirect.h" reporting is not needed for IndirectStruct1 typedef,
```

Comment:
I've chased around for quite a while trying to unravel the typedef chain here :-)

Is it possible to expand in this comment on which typedef maps to which type and _why_ reporting is needed/expected, without it turning into a mega-comment?

I think the one that stands out to me is `DoesNotForwardDeclareAndIncludes::Type ic2;` -- the general rule is that if the file declaring typedef included the type, it takes responsibility so the user of the typedef doesn't need to. Why is this nested type a special case?

Reply:
> Is it possible to expand in this comment on which typedef maps to which type and why reporting is needed/expected, without it turning into a mega-comment?

I can try to do it. It would make sense.
> Why is this nested type a special case?

No, it isn't a special case. `DoesNotForwardDeclareAndIncludes` typedef still provides its underlying type info (`DirectStruct2`), but `DirectStruct2` itself (more precisely, `DirectStruct2::Type`) doesn't provide `IndirectClass` type info.

Reply:
> No, it isn't a special case. DoesNotForwardDeclareAndIncludes typedef still provides its underlying type info 
> (DirectStruct2), but DirectStruct2 itself (more precisely, DirectStruct2::Type) doesn't provide
>  IndirectClass type info.

Ah, I didn't get that it mapped to `DirectStruct2`. That file has a forward-decl of `IndirectClass` that pushes responsibility to the use location, so it behaves as expected.

But the typedef name there is confusing everything: `DoesNotForwardDeclareAndIncludes` -- it _does_ forward-declare this new used type.

Is there another `Direct` type you could put the forward-declared use of `IndirectClass`, or does that shake everything up?

Reply:
Maybe, it's really worth to introduce separate structs and typedefs, not trying to reuse types from other test cases.